### ISSUE TITLE
const is assumed to have function scope

### ISSUE THREAD

**Issue by pornel** [jshint/jshint#1468]
```js
/* jshint esnext: true */

if (confirm("whatever")) {
  const foo = 1; alert(foo);
} else {
  const foo = 2; alert(foo);
}
```

The following warns `const 'foo' has already been declared`. However, in ES6 `const` has block scope (like `let`), so this code is correct.

**Comment by lukeapage** [comment]
I am working on a pr for this

**Comment by lukeapage** [comment]
@jugglinmike  this is fixed on master now

**Comment by jugglinmike** [comment]
Confirmed--resolved via gh-2326 / 1bb80f9a5551a639db2be0dea655c9edef7a9b8a . Expect this in the next release of JSHint (version > 2.7.0).
